File: SAMPLES/PYTHON/test_exceptions.py
# test_exceptions.py - Test TODO 21: GESTION DES EXCEPTIONS
# Compile : PYTHON86 test_exceptions.py test_exceptions.asm

# try/except simple
try:
    x = 10
    print(x)
except:
    print(0)

# try/except avec raise
try:
    raise 42
except:
    print(1)

# try/except/finally
try:
    y = 20
    print(y)
except:
    print(0)
finally:
    print(99)

# try avec raise et finally
try:
    raise 7
except:
    print(2)
finally:
    print(88)

# try/except avec type et as
try:
    raise 55
except Exception as e:
    print(e)

# raise sans expression dans try
try:
    raise
except:
    print(3)

File: SAMPLES/KOTLIN/test_exceptions.kt
// test_exceptions.kt - TODO 23 : Exceptions (try/catch/finally/throw)

// Test 1 : try/catch simple
fun main() {
    // try/catch basique
    try {
        println(42)
        throw Exception("erreur de test")
    } catch (e: Exception) {
        println(99)
    }

    // try/catch/finally
    try {
        println(1)
    } catch (e: Exception) {
        println(0)
    } finally {
        println(2)
    }

    // throw avec type specifique
    try {
        throw IllegalArgumentException("argument invalide")
    } catch (e: IllegalArgumentException) {
        println(3)
    }

    // require (ne leve pas si condition vraie)
    require(1 == 1)

    // check (ne leve pas si condition vraie)
    check(2 == 2)

    // try comme expression
    val x = try { 10 } catch (e: Exception) { 0 }
    println(x)
}
